docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,958 @@
+| `ControlPlane.ReadyReplicas` (new)       | `ControlPlane.ReadyReplicas`                             | `ControlPlane.ReadyReplicas`               |
+| `ControlPlane.UpToDateReplicas` (new)    | `ControlPlane.UpToDateReplicas`                          | `ControlPlane.UpToDateReplicas`            |
+| `ControlPlane.AvailableReplicas` (new)   | `ControlPlane.AvailableReplicas`                         | `ControlPlane.AvailableReplicas`           |
+| `Workers` (new)                          | `Workers`                                                | `Workers`                                  |
+| `Workers.DesiredReplicas` (new)          | `Workers.DesiredReplicas`                                | `Workers.DesiredReplicas`                  |
+| `Workers.Replicas` (new)                 | `Workers.Replicas`                                       | `Workers.Replicas`                         |
+| `Workers.ReadyReplicas` (new)            | `Workers.ReadyReplicas`                                  | `Workers.ReadyReplicas`                    |
+| `Workers.UpToDateReplicas` (new)         | `Workers.UpToDateReplicas`                               | `Workers.UpToDateReplicas`                 |
+| `Workers.AvailableReplicas` (new)        | `Workers.AvailableReplicas`                              | `Workers.AvailableReplicas`                |
+| other fields...                          | other fields...                                          | other fields...                            |
+
+notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes.
+
+##### Cluster (New)Conditions
+
+| Condition                 | Note                                                                                                                                                                                                                                                                                                                  |
+|---------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Available`               | True if Cluster `RemoteConnectionProbe` is true, if Cluster's control plane `Available` condition is true, if all MachineDeployment and MachinePool's `Available` condition are true; if conditions are defined in `spec.availabilityGates`, those conditions should be true as well for the Cluster to be available. |
```

Comment:
Consider a Cluster with 3 MachineDeployments, each in a separate failure domains, intended to tolerate the loss of any one of those failure domains.

> all MachineDeployment and MachinePool's `Available` condition are true

I think it'd be confusing for that Cluster to report `Available: false` when one failure domain goes down. It seems helpful to be able to distinguish between "a failure domain is down and my cluster is still available as I expected" (`Available: true`?) and "outage of one failure domain has caused some cascading issue causing the cluster to fail" (`Available: false`?).

Reply:
Not sure how we would model this :)

Reply:
Agree with @sbueringer 
Also, adding a little bit on top, we have several issues in our backlog about improving failure domains, and I think we should look at this area in a more holistic way addressing not only availability, but also rebalancing, removal/addition of failure domains, maintenance more, probably more (which is something out of the scope of this proposal).

Reply:
Resolving, eventually we can follow up on failure domain improvements on a separated issue (or on one of the existing issues about the same topic)